### PR TITLE
Add support link messages.

### PR DESCRIPTION
**Pull request by amine3**

### CODE REVIEW

File: tests/test_agent.py [hunk 1]
```diff
@@ -90,3 +90,66 @@ def virustotal_invalid_response(message):
...
+):
+    """Unittest for the lifecyle of the virustotal agent :
+    Sends a dummy malicious file through the Virus Total public API,
+    receives a valid response, assign a risk rating, creates a technical detail
+    and finally emits a message of type v3.report.vulnerability with the details above.
+    """
+
+    def virustotal_valid_response(url, timeout):
+        """Method for mocking the Virus Total public API valid response."""
+        del url, timeout
```
[the remark this answers is not in the document]
Comment (amine3) [reply tests/test_agent.py:106]: no they can't, the signature should be aligned with the one mocked

File: tests/test_agent.py [hunk 2]
```diff
@@ -90,3 +90,61 @@ def virustotal_invalid_response(message):
...
+        }
+        return response
+
+    mocker.patch(
+        "virus_total_apis.PublicApi.get_url_report",
+        side_effect=virustotal_valid_response,
+    )
+
+    virustotal_agent.process(url_message)
+    assert len(agent_mock) == 1
```
Comment (deadly-panda) [inline tests/test_agent.py:138]:
```suggestion
    virustotal_agent.process(url_message)

    assert len(agent_mock) == 1
```

File: agent/virustotal.py [hunk 3]
```diff
@@ -14,7 +16,7 @@ class VirusTotalApiError(Error):
     """VirtualTotalApiError."""
 
 
-def scan_file_from_message(message: msg.Message, api_key: str) -> Dict:
+def scan_file_from_message(message: msg.Message, api_key: str) -> dict:
```
Comment (deadly-panda) [inline agent/virustotal.py:19]: I think this will crash no ? `dict[...]`, apply to the one below also please!

**Review by deadly-panda — DISMISSED** [review]

File: agent/process_scans.py [hunk 4]
```diff
@@ -1,32 +1,38 @@
...
+    scans: dict[str, Any]
+) -> agent_report_vulnerability_mixin.RiskRating:
     """Assign risk level based on scanned file report.
 
     Returns:
         'HIGH' if at least one anti-virus detected the file as a virus, else Secure.
     """
     for scanner_result in scans.values():
-        if scanner_result["detected"]:
+        if scanner_result["detected"] is True:
```
Comment (oussamaessaji) [inline agent/process_scans.py:17]:
```suggestion
        if scanner_result.get("detected") is True:
```

File: agent/virus_total_agent.py [hunk 5]
```diff
@@ -42,26 +44,79 @@ def process(self, message: msg.Message) -> None:
...
-                entry=kb.KB.VIRUSTOTAL_SCAN,
-                technical_detail=technical_detail,
-                risk_rating=risk_rating,
-            )
+            if scans is not None:
+                technical_detail = process_scans.get_technical_details(scans, target)
+                risk_rating = process_scans.get_risk_rating(scans)
+                self.report_vulnerability(
+                    entry=kb.KB.VIRUSTOTAL_SCAN,
+                    technical_detail=technical_detail,
```
Comment (oussamaessaji) [inline agent/virus_total_agent.py:69]: Are we sure we want to report empty technical_detail sometime,

File: agent/virustotal.py [hunk 6]
```diff
@@ -30,7 +32,20 @@ def scan_file_from_message(message: msg.Message, api_key: str) -> Dict:
...
+def scan_url_from_message(target: str, api_key: str) -> dict:
+    """Method responsible for scanning a file through the Virus Total public API.
+    Args:
+        target: url to scan.
+        api_key : Key for the virustotal api.
+    Returns:
+        response: The response of the Virus Total public API.
+    """
+    virustotal_client = virus_total_apis.PublicApi(api_key)
+    response = virustotal_client.get_url_report(target, timeout=TIMEOUT_REQUEST)
```
Comment (oussamaessaji) [inline agent/virustotal.py:44]: I'm a bit concerned about the possibility of the API to crash due to the API rate limit

File: agent/virus_total_agent.py [hunk 7]
```diff
@@ -42,26 +44,79 @@ def process(self, message: msg.Message) -> None:
...
+        elif self.args.get("https") is True:
+            return "https"
+        else:
+            return "http"
+
+    def _prepare_targets(self, message: msg.Message) -> list[str]:
+        """Prepare targets based on type, if a domain name is provided, port and protocol are collected
+        from the config."""
+        if message.data.get("host") is not None:
+            host = str(message.data.get("host"))
```
[the remark this answers is not in the document]
Comment (amine3) [reply agent/virus_total_agent.py:91]: Pylint/Mypy request

File: agent/virus_total_agent.py (hunk 5 above)
Comment (amine3) [reply agent/virus_total_agent.py:69]: When will the technical details be empty?

File: agent/process_scans.py [hunk 8]
```diff
@@ -1,32 +1,38 @@
...
             return agent_report_vulnerability_mixin.RiskRating.HIGH
-    return agent_report_vulnerability_mixin.RiskRating.HIGH
+    return agent_report_vulnerability_mixin.RiskRating.SECURE
```
Comment (3asm) [inline agent/process_scans.py:19]: Ouch!, good catch.

File: agent/process_scans.py [hunk 9]
```diff
@@ -1,32 +1,38 @@
...
 
     Returns:
         technical_detail : Markdown table of the scans results.
     """
     scans = markdown.prepare_data_for_markdown_formatting(scans)
-    technical_detail = markdown.table_markdown(scans)
+    technical_detail = ""
+    if target is not None:
+        technical_detail = f"Analysis of the target `{target}`:\n"
+    technical_detail += markdown.table_markdown(scans)
```
Comment (3asm) [inline agent/process_scans.py:37]: What table will be set if the target is missing?

File: agent/virus_total_agent.py [hunk 10]
```diff
@@ -42,26 +44,79 @@ def process(self, message: msg.Message) -> None:
...
+    def _get_schema(self, message: msg.Message) -> str:
+        """Returns the schema to be used for the target."""
+        if message.data.get("schema") is not None:
+            return str(message.data["schema"])
+        elif message.data.get("protocol") is not None:
+            return str(message.data["protocol"])
+        elif self.args.get("https") is True:
+            return "https"
+        else:
+            return "http"
```
Comment (3asm) [inline agent/virus_total_agent.py:85]: Nice.

**Review by 3asm — APPROVED** [review]

**Pull request merged**